ipa-core/src/protocol/ipa_prf/boolean_ops/share_conversion_aby.rs

```diff
@@ -280,10 +305,11 @@ where
 /// NP records.
 fn output_shares<C, const NC: usize, const NP: usize>(
     ctx: &C,
+    results: &mut Vec<AdditiveShare<Fp25519, NP>>,
```

Comment:
I suggested it in `https://github.com/private-attribution/ipa/pull/1180#discussion_r1664585805`.

Previously, `output_shares` created a vector, returned it, and it was then passed through as the return value of `convert_to_fp25519`. This PR is changing `convert_to_fp25519` to operate over larger chunks, which meant that it was assembling the vectors returned by `output_shares` into a larger vector. Rather than assemble small vectors and then immediately drain those smaller vectors into a larger vector, it seemed better to just assemble the output in the larger vector directly.